lib/routes/user/marketplace/vendor_webview.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:async';
+import 'dart:convert' as JSON;
+import 'package:flutter/material.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/invoice/invoice_bloc.dart';
+import 'package:flutter_webview_plugin/flutter_webview_plugin.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/loader.dart';
+
+class VendorWebViewPage extends StatefulWidget {
+  final String _url;
+  final String _title;
+
+  VendorWebViewPage(this._url, this._title);
+
+  @override
+  State<StatefulWidget> createState() {
+    return new VendorWebViewPageState();
+  }
+}
+
+class VendorWebViewPageState extends State<VendorWebViewPage> {
+  StreamSubscription _postMessageListener;
+  final _widgetWebview = new FlutterWebviewPlugin();
+
+  @override
+  void initState() {
+    super.initState();
+    _widgetWebview.onStateChanged.listen((state) async {
+      if (state.type == WebViewState.finishLoad) {
+        String script =
+            'window.addEventListener("message", receiveMessage, false);' +
+                'function receiveMessage(event) {Android.getPostMessage(event.data);}';
+        _widgetWebview.evalJavascript(script);
+      }
+    });
+
+    _widgetWebview.onDestroy.listen((_) {
+      if (Navigator.canPop(context)) {
+        Navigator.of(context).pop();
+      }
+    });
+  }
+
+  @override
+  void didChangeDependencies() {
+    InvoiceBloc invoiceBloc = AppBlocsProvider.of<InvoiceBloc>(context);
+
+    _postMessageListener = _widgetWebview.onPostMessage.listen((postMessage) {
```

Comment:
didChangeDependencies may be called multiple times during the widget lifecycle. You should make sure the _postMessageListener is only initialized once and can do that by using an _init flag.